docs/proposals/policy.md

```diff
@@ -0,0 +1,438 @@
+	"name": "Policy",
+	"namespace": "hammer"
+}
+{
+	"kind": "PolicyBinding",
+	"name": "hammer",
+	"namespace": "hammer"
+}
+```
+
+Once those container resources are created, `Hubert` can now create Roles and RoleBindings that use them.
+
+`Hubert` wants to create a bot that only has permission to touch labels on DeploymentConfigs.  He can create a role like this (note that a role is required because we want to have a set of two rules):
+```
+{
+	"kind": "role",
+	"name": "deploymentConfigLabelers",
+	"namespace": "hammer",
+	"rules": [
+		{ "verbs": ["watch", "list", "get"], "resourceKinds": ["DeploymentConfig"] },
+		{ "verbs": ["update"], "resourceKinds": ["DeploymentConfig"], "attributeRestrictions" : {"fieldsMutatable": ["labels"]} }
+	]
+}
+{
+	"kind": "roleBinding",
+	"name": "DeploymentConfigLabelerBots",
+	"namespace": "hammer",
+	"roleRef": {
+		"namespace": "hammer",
+		"name": "deploymentConfigLabelers"
+	}
+	"userNames": ["ProtectorBot", "DeprotectorBot"]
+}
+```
+
+Finally, `Hubert` wants to prevent `Edgar` from ever deleting a protected deploymentConfig again, so he creates this role and roleBinding:
+```
+{
+	"kind": "role",
+	"name": "fatFingeredEditor",
+	"namespace": "hammer",
+	"rule": [
+		{ "Deny": "true", "Verbs": ["delete"], "ResourceKinds": ["DeploymentConfig"], "AttributeRestrictions" : {"labelsContain": ["protected"]} }
+	]
+}
+{
+	"kind": "roleBinding",
+	"name": "FatFingeredEditors",
+	"namespace": "hammer",
+	"roleRef": {
+		"namespace": "hammer",
+		"name": "fatFingeredEditor"
+	}
+	"userNames": ["Edgar"]
+}
+```
+
+
+#### Kubelets can modify Pods that they are running, but no others
+In addition to constraining basic resources, there are some rules that are more difficult to request.  For some of these, it is easier to build a special evaluation method than trying to specify a generic set of property to express the restriction.  One way to express this is to create a rule for each individual pod that allows that particular kubelet use to access it.  This results in a tremendous number of rules and a lot of churn in the rules themselves.  Rather than do this, we can add a special kind of AttributeRestriction to a single rule.
+```
+{
+	"kind": "role",
+	"name": "kubelet",
+	"namespace": "master",
+	"rules": [
+		{
+			"verbs": ["*"], 
+			"resourceKinds": ["Pod"],  
+			"attributeRestrictions": {
+				"kind": "sameMinionRestriction"
+			}
+		},
+	]
+}
+{
+	"kind": "roleBinding",
+	"name": "Kubelets",
+	"namespace": "master",
+	"roleRef": {
+		"namespace": "master",
+		"name": "kubelet"
+	}
+	"userNames": ["kubelet-01", "kubelet-02"]
+}
+```
+
+When Authorizer.Authorize() is called, it will see a 
+```
+// pretend authorizationAttributes
+{
+	"user": "kubelet",
+	"verb": "get",
+	"namespace": "pod's namespace",
+	"resourceKind": "pod"
+}
+
+// kubelet policy rule it extracted based on the role
+{
+	"verbs": ["*"], 
+	"resourceKinds": ["Pod"],  
+	"attributeRestrictions": {
+		"kind": "sameMinionRestriction"
```

Comment:
sameNodeRestriction would be more appropriate with the rename of Minion to Node.